homeassistant/components/islamic_prayer_times/__init__.py

```diff
@@ -155,7 +154,11 @@ async def async_setup(self):
         await self.async_update()
         self.config_entry.add_update_listener(self.async_options_updated)
 
-        self.hass.config_entries.async_setup_platforms(self.config_entry, PLATFORMS)
+        self.hass.data.setdefault(DOMAIN, self)
```

Comment:
I had to move this because the platform would try to use `hass.data[DOMAIN]` but it wasn't set yet

Reply:
I think the `setdefault` is an anti pattern since this is a singleton config entry, and we unconditionally pop it in `async_unload_entry`
```suggestion
```